### PR TITLE
migrate some file to null safety

### PR DESCRIPTION
**Pull request by wanglikun7342**
migrate some file to null safety，list
mirgate integration.shard/debugger_stepping_test.dart to null safety
mirgate integration.shard/coverage_collection_test.dart to null safety
mirgate integration.shard/command_output_test.dart to null safety
mirgate integration.shard/build_ios_config_only_test.dart to null safety
mirgate integration.shard/background_isolate_test.dart to null safety
mirgate integration.shard/android_plugin_example_app_build_test.dart to null safety
mirgate integration.shard/analyze_size_test.dart to null safety
migrate integration.shard/analyze_once_test.dart to null safety
mirgate debug_adapter/flutter_adapter_test.dart to null safety

*List which issues are fixed by this PR. You must list at least one issue.*
#71511

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*
NO

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by wanglikun7342** [comment]
@jmagman null safe mirgate pr

File: packages/flutter_tools/test/integration.shard/command_output_test.dart [hunk 1]
```diff
@@ -138,11 +136,11 @@ void main() {
       '--machine',
     ]);
 
-    final Map<String, Object> versionInfo = json.decode(result.stdout
+    final Map<String, dynamic> versionInfo = json.decode(result.stdout
       .toString()
       .replaceAll('Building flutter tool...', '')
       .replaceAll('Waiting for another flutter command to release the startup lock...', '')
-      .trim()) as Map<String, Object>;
+      .trim()) as Map<String, dynamic>;
```
Comment (jmagman) [inline packages/flutter_tools/test/integration.shard/command_output_test.dart:143]: These should be `Object` or `Object?` not `dynamic`.  `dynamic` removes static type checking and shouldn't be used in null safe libraries.
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-var-and-dynamic

**Review by jmagman — CHANGES_REQUESTED** [review]

**Comment by wanglikun7342** [comment]
i think the check errors  make mistake, jmagman. The files I modified is irrelevant.

**Comment by jmagman** [comment]
> i think the check errors make mistake, jmagman. The files I modified is irrelevant.

You're right, it's https://github.com/flutter/flutter/issues/93133. That will be fixed soon, and then you can rebase onto the fix and the tests should pass.

**Review by jmagman — APPROVED** [review]
LGTM

File: packages/flutter_tools/test/integration.shard/debug_adapter/flutter_adapter_test.dart [hunk 2]
```diff
@@ -119,7 +117,7 @@ void main() {
     await _project.setUpIn(tempDir);
 
     // Launch the app and wait for it to print "topLevelFunction".
-    await Future.wait(<Future<Object>>[
+    await Future.wait(<Future<void>>[
```
Comment (christopherfujino) [inline packages/flutter_tools/test/integration.shard/debug_adapter/flutter_adapter_test.dart:120]: why was this changed to `void`?

**Comment by christopherfujino** [comment]
@wanglikun7342 are you still working on this?

**Comment by wanglikun7342** [comment]
@christopherfujino I have resolved it , please review it again

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 10.
View them at https://flutter-gold.skia.org/cl/github/92957

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 10.
View them at https://flutter-gold.skia.org/cl/github/92957

**Review by christopherfujino — APPROVED** [review]
LGTM thanks!

**Pull request merged**